File: aramis-opengl/src/main/java/hello/com/aramis/opengl/december/views/DouyinView.java
package hello.com.aramis.opengl.december.views;

import android.content.Context;
import android.opengl.GLSurfaceView;
import android.util.AttributeSet;
import android.view.SurfaceHolder;

/**
 * Created by Aramis
 * Date:2018/12/5
 * Description:
 */
public class DouyinView extends GLSurfaceView {
    private DouyinRender douyinRender;
    private Speed mSpeed = Speed.MODE_NORMAL;

    public DouyinView(Context context) {
        super(context);
    }

    public DouyinView(Context context, AttributeSet attrs) {
        super(context, attrs);
        douyinRender = new DouyinRender(this);
        setEGLContextClientVersion(2);
        setRenderer(douyinRender);
        //设置为按需渲染
        setRenderMode(RENDERMODE_WHEN_DIRTY);
    }

    public void autoFocus() {
        douyinRender.autoFocus();
    }

    @Override
    public void surfaceDestroyed(SurfaceHolder holder) {
        super.surfaceDestroyed(holder);
        douyinRender.onSurfaceDestroyed();
    }

    public void startRecord() {
        float speed = 1.0f;
        switch (mSpeed) {
            case MODE_EXTRA_SLOW:
                speed = 0.3f;
                break;
            case MODE_SLOW:
                speed = 0.5f;
                break;
            case MODE_FAST:
                speed = 1.5f;
                break;
            case MODE_EXTRA_FAST:
                speed = 3.0f;
                break;
        }
        douyinRender.startRecord(speed);
    }

    public void stopRecord() {
        douyinRender.stopRecord();
    }

    public void setSpeed(Speed speed) {
        this.mSpeed = speed;
    }

    public enum Speed {
        MODE_EXTRA_SLOW, MODE_SLOW, MODE_NORMAL, MODE_FAST, MODE_EXTRA_FAST
    }
}

File: aramis-opengl/src/main/java/hello/com/aramis/opengl/douyin/DouyinView.kt
package hello.com.aramis.opengl.douyin

import android.content.Context
import android.opengl.GLSurfaceView
import android.util.AttributeSet
import android.view.SurfaceHolder

/**
 *Created by Aramis
 *Date:2018/10/31
 *Description:
 */
class DouyinView : GLSurfaceView {
    private val renderer=DouyinRender(this)
    fun init(attrs: AttributeSet?) {
        //设置EGL版本
        setEGLContextClientVersion(2)
        setRenderer(renderer)
        //RENDERMODE_WHEN_DIRTY:按需渲染
        //RENDERMODE_CONTINUOUSLY:一直渲染
        renderMode= RENDERMODE_WHEN_DIRTY

    }

    override fun surfaceDestroyed(holder: SurfaceHolder?) {
        super.surfaceDestroyed(holder)
        renderer.onSurfaceDestroy()
    }



    constructor(context: Context?) : super(context) {
        init(null)
    }

    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs) {
        init(attrs)
    }
}